### DISCUSSION TITLE
Possible to bypass Istio ServiceMesh in Kubernetes by IP Address?

### DISCUSSION THREAD

**Discussion by NitishKumar525** [istio/istio#53031] [Q&A] [answered]
We're using a Service Mesh with Istio for things like mTLS between the services. What happens when I address a POD directly by its IP address. Would this work to bypass things we set up with Istio? From my understanding, using the IP address from the Pod where the application is running, will ultimately bypass the service mesh?

**Comment by mdazfar2** [comment, marked as answer]
No it will not bypass Istio.

The pod is configured with an IP address, that IP address will direct traffic to the [Istio Sidecar](https://istio.io/latest/docs/reference/config/networking/sidecar/) container in the pod which will in turn call the service container on the pod's localhost.

Good explanation [here](https://pramodshehan.medium.com/istio-service-mesh-a7b8204f9da2#:%7E:text=A%20sidecar%20is%20a%20separate,protocols%20such%20as%20HTTP%2C%20GRPC.).

**Comment by howardjohn** [comment]
See https://istio.io/latest/docs/ops/configuration/traffic-management/traffic-routing/#headless-services (the whole page really, but especially the part I linked).

tl;dr: usually it will not get full mesh functionality. it still goes through the proxy. This limitation does not apply to ambient mode (see https://blog.howardjohn.info/posts/securing-prometheus/; prometheus is just an example)
